### PR TITLE
Feature/upgrades

### PR DESCRIPTION
**Pull request by lindenmckenzie**
### What

Upgrade Go / Node and package depdencies
Add pr-audit and pr-unit tests

### How to review

Run make watch-serve and see that it runs and renders things correctly. 
Check npm commands still work

### Who can review

Not me.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,12 +10,14 @@ To run this app you'll need certain versions installed:
 
 | Language | version   |
 |----------|-----------|
-| Go       | >= 1.20.3 |
-| Node     | = 14.21.3 |
+| Go       | >= 1.21.7 |
```
Comment (gedge) [inline README.md:13]: `go1.22.5` is a thing...

File: main.go [hunk 2]
```diff
@@ -427,7 +435,7 @@ func generateStaticMetadata(md []byte) (b []byte, metadata map[string]string) {
 func generateStyledCodeHTML(html []byte) []byte {
 	doc, err := goquery.NewDocumentFromReader(bytes.NewReader(html))
 	if err != nil {
-		log.Event(nil, "Failed to read html file", log.Error(err))
+		log.Error(context.TODO(), "Failed to read html file", err)
```
Comment (gedge) [inline main.go:438]:
```suggestion
		log.Error(context.TODO(), "failed to read html file", err)
```
you stopped lower-casing error messages a while back

File: Makefile [hunk 3]
```diff
@@ -42,5 +44,5 @@ watch-serve:
 
 .PHONY: serve
 serve:
-	go install github.com/fogleman/serve
+	go install github.com/fogleman/serve@latest
```
Comment (gedge) [inline Makefile:47]: I believe that nothing should be (surprise) installed as part of a `make serve` (or similar).

That should be in the prerequisites.

Admittedly, `ci` needs it, but that can be done in the `ci/scripts` explicitly

Comment (lindenmckenzie) [reply Makefile:47]: mmm I did think something along these lines, it's not needed for `ci` though I think as we just build the flat files? 

What would your recommended approach be? Have it in the pre-reqs in the readme and then have it fail if they don't have it?

Comment (gedge) [reply Makefile:47]: in that case, yes (pre-reqs and fail)

Comment (gedge) [reply Makefile:47]: could move the install line to (e.g) `make install-pre-reqs` or similar

**Review by gedge — APPROVED** [review]

**Pull request merged**